file-formats/tabl/type/zif_aff_indx_v1.intf.abap

```diff
@@ -0,0 +1,42 @@
+INTERFACE zif_aff_indx_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Index Fields</p>
+    "! Index fields
+    BEGIN OF ty_indexfield,
+      "! <p class="shorttext">Field Name</p>
+      "! Field name
+      fieldname TYPE c LENGTH 30,
+      "! <p class="shorttext">Field Position</p>
+      "! Position
+      position  TYPE n LENGTH 4,
+    END OF ty_indexfield,
+    tt_index_fields TYPE SORTED TABLE OF ty_indexfield WITH UNIQUE KEY fieldname.
+
+  TYPES:
+    "! <p class="shorttext">ABAP DDIC Secondary Index Properties</p>
+    "! Object type INDX
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Text Index</p>
+      "! Standard- or text-Index
+      "! $required
+      "! $default 'abap_false'
+      text_index     TYPE abap_bool,
```

Comment:
That is in principle a good idea. The question is how this will look afterwards on the GUI. So far we are much in favor of a checkbox which only must be selected if the developer actually wants a text index (should be the 1% case). I guess if the change this to an enumeration we cannot have a checkbox, right?  
The question is also related to the support of old-style fulltext indexes. That would require the ability to enable and disable certain fields based on the index type. I think so far we said that this is not possible.

Reply:
It would become a drop-down with both options. The "old-style fulltext indexes" would be a third option, wouldn't it?

For the ABAP file formats it is most probably helpful to specify the full object type even if we use/support just a subset in ADT in a first version.